provider/azure.go

```diff
@@ -337,6 +351,10 @@ func (p *AzureProvider) deleteRecords(ctx context.Context, deleted azureChangeMa
 	for zone, endpoints := range deleted {
 		for _, endpoint := range endpoints {
 			name := p.recordSetNameForZone(zone, endpoint)
+			if len(p.zoneNameFilter.filters) > 0 && !p.domainFilter.Match(endpoint.DNSName) {
```

Comment:
wouldn't this be enough to only check !p.domainFilter.Match(endpoint.DNSName)?
same below

Reply:
fixed